estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,114 @@
+        string="Status",
+        copy=False,
+    )
+    partner_id = fields.Many2one("res.partner", string="Partner", required=True)
+    property_id = fields.Many2one(
+        "estate.property",
+        string="Property",
+        required=True,
+        ondelete="cascade",
+    )
+    property_type_id = fields.Many2one(
+        comodel_name="estate.property.type",
+        string="Property Type",
+        related="property_id.property_type_id",
+        store=True,
+    )
+
+    validity = fields.Integer(string="Validity (days)", default=7)
+    date_deadline = fields.Date(string="Deadline", compute="_compute_date_deadline", inverse="_inverse_date_deadline", store=False)
+
+    @api.depends("validity", "create_date")
+    def _compute_date_deadline(self) -> None:
+        for offer in self:
+            created = (offer.create_date or fields.Datetime.now())
+            base_date = created.date() if hasattr(created, "date") else date.today()
+            offer.date_deadline = base_date + timedelta(days=offer.validity or 0)
+
+    def _inverse_date_deadline(self) -> None:
+        for offer in self:
+            created = (offer.create_date or fields.Datetime.now())
+            base_date = created.date() if hasattr(created, "date") else date.today()
+            if offer.date_deadline:
+                delta = offer.date_deadline - base_date
+                offer.validity = max(delta.days, 0)
+
+    # Make the form reactive while editing (no need to wait for save)
+    @api.onchange("validity")
+    def _onchange_validity(self):
+        for offer in self:
+            created = (offer.create_date or fields.Datetime.now())
+            base_date = created.date() if hasattr(created, "date") else date.today()
+            offer.date_deadline = base_date + timedelta(days=offer.validity or 0)
+
+    @api.onchange("date_deadline")
+    def _onchange_date_deadline(self):
+        for offer in self:
+            if offer.date_deadline:
+                created = (offer.create_date or fields.Datetime.now())
+                base_date = created.date() if hasattr(created, "date") else date.today()
+                delta = offer.date_deadline - base_date
+                offer.validity = max(delta.days, 0)
+
+    # Actions on offers
+    def action_accept(self):
+        for offer in self:
+            if offer.property_id.state == "sold":
+                raise UserError("Cannot accept an offer on a sold property.")
+            # set other offers to refused
+            siblings = offer.property_id.offer_ids - offer
+            siblings.write({"status": "refused"})
+            offer.status = "accepted"
+            offer.property_id.write({
+                "buyer_id": offer.partner_id.id,
+                "selling_price": offer.price,
+                "state": "offer_accepted",
+            })
+        return True
+
+    def action_refuse(self):
+        self.write({"status": "refused"})
```

Comment:
```suggestion
        self.status = "refused"
```
It is better to let the orm decide when the write should be applied